app/models/user.rb

```diff
@@ -4,13 +4,16 @@ class User < ApplicationRecord
 
   has_many :clients
   has_one :profile, dependent: :destroy
+  has_one :volunteer, dependent: :destroy
   has_and_belongs_to_many :department
 
   # Roles definition
   SUPERADMIN = 'superadmin'.freeze
   SOCIAL_WORKER = 'social_worker'.freeze
   DEPARTMENT_MANAGER = 'department_manager'.freeze
-  ROLES = [SUPERADMIN, SOCIAL_WORKER, DEPARTMENT_MANAGER].freeze
+  VOLUNTEER = 'volunteer'.freeze
+  ROLES_FOR_COLLECTION = [SUPERADMIN, SOCIAL_WORKER, DEPARTMENT_MANAGER].freeze
```

Comment:
You're right. Since volunteers will have an account, therefore access to the system, I would suggest ROLES_FOR_USER_CREATE or ROLES_FOR_CREATE.